sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -642,6 +642,13 @@ object SQLConf {
       .bytesConf(ByteUnit.BYTE)
       .createWithDefault(0L)
 
+  val ADAPTIVE_COST_EVALUATOR_CLASS =
+    buildConf("spark.sql.adaptive.costEvaluatorClass")
+      .version("3.2.0")
+      .internal()
+      .stringConf
+      .createWithDefault("org.apache.spark.sql.execution.adaptive.SimpleCostEvaluator")
```

Comment:
We can make it an optional conf: `spark.sql.adaptive.customCostEvaluatorClass`. If not set, we use the builtin impl.

Reply:
@cloud-fan - sure, updated.